builtin/method_str.py

```diff
@@ -488,38 +488,89 @@ def __init__(self):
     def Call(self, rd):
         # type: (typed_args.Reader) -> value_t
         """
-        s.split(sep, count=-1)
+        s.split(delim, count=-1)
+        s.split(eggex, count=-1)
 
         Count behaves like in replace() in that:
         - `count` <  0 -> ignore
         - `count` >= 0 -> there will be at most `count` splits
         """
         string = rd.PosStr()
-        sep = rd.PosStr()
+
+        string_sep = None  # type: str
+        eggex_sep = None  # type: value.Eggex
+
+        sep = rd.PosValue()
+        with tagswitch(sep) as case:
+            if case(value_e.Eggex):
+                eggex_sep_ = cast(value.Eggex, sep)
+                eggex_sep = eggex_sep_
+
+            elif case(value_e.Str):
+                string_sep_ = cast(value.Str, sep)
+                string_sep = string_sep_.s
+
+            else:
+                raise error.TypeErr(sep, 'expected sep to be Eggex or Str',
```

Comment:
I think in user facing text it's worth it to spell out "separator"

JS and Python seem to call it "separator" rather than "delimiter"

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/split